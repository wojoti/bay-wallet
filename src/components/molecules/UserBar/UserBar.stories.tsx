import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {UserBar} from 'components/molecules';

export default {
  title: 'Molecules/UserBar',
  component: UserBar,
  argTypes: {
    onNotificationPress: {action: 'onNotificationPress'},
    onUserProfilePress: {action: 'onNotificationPress'},
  },
} as ComponentMeta<typeof UserBar>;
const Template: ComponentStory<typeof UserBar> = args => <UserBar {...args} />;

export const Example = Template.bind({});
Example.args = {
  testId: '',
  username: 'David Kowalski',
};
