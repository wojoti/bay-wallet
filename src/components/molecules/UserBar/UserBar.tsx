import {
  Container,
  EmptyButton,
  Header,
  Img,
  ImgType,
  Label,
} from 'components/atoms';
import {PropsWithChildren} from 'react';
import {secondary} from 'theme/colors';

export type UserBarProps = PropsWithChildren<{
  username: string;
  onNotificationPress: () => void;
  onUserProfilePress: () => void;
  testId?: string;
}>;

const UserBar = ({
  username,
  onNotificationPress,
  onUserProfilePress,
  testId,
}: UserBarProps) => {
  return (
    <Container direction="row" testId={testId}>
      <Container align="flex-start" flex={2.6}>
        <Label testId="userbar-welcome" color={secondary}>
          Good Morning
        </Label>
        <Header testId="userbar-username" color={secondary}>
          {username}
        </Header>
      </Container>
      <Container
        justify="space-between"
        flex={1}
        align="center"
        direction="row">
        <EmptyButton
          onPress={onNotificationPress}
          testId="userbar-notification">
          <Img src={ImgType.notificationRed} height={26} width={24} alt={''} />
        </EmptyButton>
        <EmptyButton onPress={onUserProfilePress} testId="userbar-userprofile">
          <Img src={ImgType.profile} height={40} width={40} alt={''} />
        </EmptyButton>
      </Container>
    </Container>
  );
};

export default UserBar;
