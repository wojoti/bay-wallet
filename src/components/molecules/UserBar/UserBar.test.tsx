import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import UserBar, {UserBarProps} from './UserBar';

const mockOnNotificationPress = jest.fn();
const mockOnUserProfilePress = jest.fn();
const props: UserBarProps = {
  testId: 'test-userbar-id',
  onNotificationPress: mockOnNotificationPress,
  onUserProfilePress: mockOnUserProfilePress,
  username: 'test username',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<UserBar {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render UserBar', () => {
  render(<UserBar {...props} />);
  const UserBarElement = screen.getByTestId('test-userbar-id');
  expect(UserBarElement).toBeOnTheScreen();
});

test('should render UserBar with welcome label', () => {
  render(<UserBar {...props} />);
  const UserBarElement = screen.getByTestId('test-userbar-id');
  expect(UserBarElement).toBeOnTheScreen();
  const UserBarLabelElement = screen.getByTestId('userbar-welcome');
  expect(UserBarLabelElement).toBeOnTheScreen();
});

test('should render UserBar with user name', () => {
  render(<UserBar {...props} />);
  const UserBarElement = screen.getByTestId('test-userbar-id');
  expect(UserBarElement).toBeOnTheScreen();
  const UserBarHeaderElement = screen.getByTestId('userbar-username');
  expect(UserBarHeaderElement).toBeOnTheScreen();
});

test('should render UserBar with working notification button', () => {
  render(<UserBar {...props} />);
  const UserBarElement = screen.getByTestId('test-userbar-id');
  expect(UserBarElement).toBeOnTheScreen();
  const UserBarNotificationElement = screen.getByTestId('userbar-notification');
  expect(UserBarNotificationElement).toBeOnTheScreen();
  expect(mockOnNotificationPress).toBeCalledTimes(0);
  fireEvent.press(UserBarNotificationElement);
  expect(mockOnNotificationPress).toBeCalledTimes(1);
});

test('should render UserBar with working user profile button', () => {
  render(<UserBar {...props} />);
  const UserBarElement = screen.getByTestId('test-userbar-id');
  expect(UserBarElement).toBeOnTheScreen();
  const UserBarUserProfileElement = screen.getByTestId('userbar-userprofile');
  expect(UserBarUserProfileElement).toBeOnTheScreen();
  expect(mockOnUserProfilePress).toBeCalledTimes(0);
  fireEvent.press(UserBarUserProfileElement);
  expect(mockOnUserProfilePress).toBeCalledTimes(1);
});
