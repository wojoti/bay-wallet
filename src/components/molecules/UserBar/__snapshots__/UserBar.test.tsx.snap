// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-userbar-id"
>
  <View
    style={
      [
        {
          "alignItems": "flex-start",
          "flex": 2.6,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "Poppins-Light",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "left",
        }
      }
      testID="userbar-welcome"
    >
      Good Morning
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "Poppins-Bold",
          "fontSize": 24,
          "textAlign": "left",
        }
      }
      testID="userbar-username"
    >
      test username
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderStyle": "solid",
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          undefined,
        ]
      }
      testID="userbar-notification"
    >
      <SvgMock
        height={26}
        width={24}
      />
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderStyle": "solid",
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          undefined,
        ]
      }
      testID="userbar-userprofile"
    >
      <SvgMock
        height={40}
        width={40}
      />
    </View>
  </View>
</View>
`;
